### PR TITLE
Update documentation and other minor changes

### PR DESCRIPTION
**Pull request by desilinguist**
- Add detailed notes about writing tests to `contributing.rst`.
- Update API page with a note about backwards compatibility. Also add functions from `rsmtool.utils`.
- Delete `ci` folder.
- Update SKLL to v1.5.1 and remove SKLL logger nullification.
- Fix a bug in tests where `file_format` was not being passed properly.

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -4,6 +4,34 @@ API Documentation
 -----------------
 The primary method of using RSMTool is via the command-line scripts :ref:`rsmtool <usage_rsmtool>`, :ref:`rsmeval <usage_rsmeval>`, :ref:`rsmpredict <usage_rsmpredict>`, :ref:`rsmcompare <usage_rsmcompare>`, and :ref:`rsmsummarize <usage_rsmsummarize>`. However, there are certain functions in the ``rsmtool`` API that may also be useful to advanced users for use directly in their Python code. We document these functions below.
 
+.. note::
+    
+    Previous versions of the RSMTool API provided the functions ``metrics_helper``, ``convert_ipynb_to_html``, and ``remove_outliers``. These functions have now been turned into static methods for different classes. If you are using these functions in your code and want to migrate to the new API, you should replace the follwing statements in your code:
```
Comment (aloukina) [inline doc/api.rst]: Add actual version number? Or `versions released before February 2018`?

File: doc/contributing.rst [hunk 2]
```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
...
+        |   field in the configuration file.                                         |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMEval                                                |
+        |                                                                            |
+        | * Same arguments as RSMTool except the ``skll`` keyword argument is not    |
+        |   applicable.                                                              |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMPredict                                             |
+        |                                                                            |
+        | * Only positional argument is the name of the test directory you created.  |
```
Comment (aloukina) [inline doc/contributing.rst]: "The" only?

File: doc/contributing.rst [hunk 3]
```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
...
+        | Writing test(s) for RSMCompare                                             |
+        |                                                                            |
+        | * First positional argument is the name of the test directory you created. |
+        |                                                                            |
+        | * Second positional argument is the comparison ID from the JSON            |
+        |   configuration file.                                                      |
+        +----------------------------------------------------------------------------+
+        | Writing test(s) for RSMSummarize                                           |
+        |                                                                            |
+        | * Only positional argument is the name of the test directory you created.  |
```
Comment (aloukina) [inline doc/contributing.rst]: "The" only?

File: doc/contributing.rst [hunk 4]
```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
...
+        |                                                                            |
+        | * Set ``file_format="tsv"`` (or ``"xlsx"``) if you specified the same      |
+        |   field in the configuration file.                                         |
+        +----------------------------------------------------------------------------+
+
+Once you have added all new functional tests, you should run ``nosetests --noglogcapture`` to run all the tests. Obviously, the newly added tests will fail since you have not yet generated the expected output for that test. To do this, you should now run the following:
```
Comment (aloukina) [inline doc/contributing.rst]: Somewhere here should be a step where one reviews the outputs and makes sure they are as expected before adding them to the test.

Comment (aloukina) [reply doc/contributing.rst]: Also, I would add a recommendation to commit changes before updating test files. In this case if updating the files went terribly wrong you can just stash them and start from the last commit.

File: doc/contributing.rst [hunk 5]
```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
...
+
+.. _update_files:
+.. code-block:: text
+    
+    python tests/update_files.py --tests tests --outputs test_outputs
+
+This will copy over the generated outputs for the newly added tests and show you a report of the files that it added. If you run ``nosetests`` again, your newly added tests should now pass. 
+
```
Comment (aloukina) [inline doc/contributing.rst]: "If run correctly, you should only see changes to model files and the files affected by the functionality you implemented"

File: doc/contributing.rst [hunk 6]
```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
...
+At this point, you should commit all of your changes included the expected test outputs that you should see under ``tests/data/experiments/<test>/output`` where ``<test>`` refers to the test(s) that you added. 
+
+Advanced tips and tricks
+------------------------
+
+Here are some advanced tips and tricks when working with RSMTool tests.
+
+1. To run a specific test function in a specific test file, simply use ``nosetests --nologcapture tests/test_X.py:Y`` where ``test_X.py`` is the name of the test file, and ``Y`` is the test functions. Note that this will not work for parameterized tests. If you want to run a specific parameterized test, you can comment out all of the other ``param()`` calls and run the ``test_run_experiment_parameterized()`` function as above.
+
+2. If you make any changes to the code that can change the output that the tests are expected to produce, you *must* re-run all of the tests and the update the expected test outputs using the ``update_files.py`` command as shown :ref:`above <update_files>`.
```
Comment (aloukina) [inline doc/contributing.rst]: the update -> then update
expected -> *expected*

File: doc/contributing.rst [hunk 7]
```diff
@@ -15,4 +20,116 @@ Contributions to RSMTool are very welcome. You can use the instructions below to
...
+
+3. In the rare case that you *do* need to create an entirely new ``tests/test_experiment_X.py`` file instead of using one of the existing ones, you can choose whether to exclude the tests contained in this file from updating their expected outputs when ``update_files.py`` is run by setting ``_AUTO_UPDATE=False`` at the top of the file. This should *only* be necessary if you are absolutely sure that your tests do not need updating.
```
Comment (aloukina) [inline doc/contributing.rst]: "do not need updating" -> "will never need updating"

File: doc/contributing.rst (hunk 4 above)
Comment (desilinguist) [reply doc/contributing.rst]: Yup, good ideas. Will add those steps.

File: doc/api.rst [hunk 8]
```diff
@@ -4,6 +4,34 @@ API Documentation
...
+
+    with the following, respectively:
+
+    .. code-block:: python
+
+        from rsmtool.analyzer import Analyzer
+        Analyzer.metrics_helper(...)
+
+        from rsmtool.reporter import Reporter
+        Reporter.convert_ipynb_to_htmlfr(...)
```
Comment (jbiggsets) [inline doc/api.rst]: Should be `Reporter.convert_ipynb_to_html(...)`, right?

Comment (desilinguist) [reply doc/api.rst]: Ugh. Yes :)

**Pull request merged**